src/error.rs

```diff
@@ -98,35 +98,11 @@ impl Error {
     }
 }
 
-cfg_if! {
-    if #[cfg(unix)] {
-        fn os_err(errno: i32, buf: &mut [u8]) -> Option<&str> {
-            let buf_ptr = buf.as_mut_ptr().cast::<libc::c_char>();
-            if unsafe { libc::strerror_r(errno, buf_ptr, buf.len()) } != 0 {
-                return None;
-            }
-
-            // Take up to trailing null byte
-            let n = buf.len();
-            let idx = buf.iter().position(|&b| b == 0).unwrap_or(n);
-            core::str::from_utf8(&buf[..idx]).ok()
```

Comment:
Even if the bytes do not contain any invalid UTF-8 sequences, they might not be encoding UTF-8.

Reply:
Is there an actual example of this being an issue on a `unix` target?

Reply:
No offense, but I don't participate in that kind of thinking. Instead it would be better to have a proof that it isn't an issue.

Reply:
Oh no offense taken, I think this is a good point (especially when it comes to stuff like the safety of code or cryptographic security where I 100% agree). Sorry for being a little short. I just meant to note that an implementation returning UTF8 bytes that aren't UTF8 only causes the display implementation to show a suboptimal error string.

However, after looking more into `strerror_r`, I agree that we shouldn't be assuming that it is well behaved on arbitrary `unix` targets. This is `libstd`'s job, not ours. See my suggestion below.